server/text_generation_server/utils/chunks.py

```diff
@@ -0,0 +1,20 @@
+from typing import Iterable
+
+from loguru import logger
+
+from text_generation_server.pb import generate_pb2
+
+
+def concat_text_chunks(chunks: Iterable[generate_pb2.InputChunk]) -> str:
+    """
+    Concatenate text in text chunks. Non-text chunks are dropped.
+    """
+    texts = []
+    for chunk in chunks:
+        chunk_type = chunk.WhichOneof("chunk")
+        if chunk_type == "text":
+            texts = chunk.text
```

Comment:
related to above, maybe we append `chunk.text` to `texts` to handle multiple text chunks?

```suggestion
            texts.append(chunk.text)
```

Reply:
Thanks, that should indeed have been an append, fixed!